### PR TITLE
Refactor to clean the main.rs and remove warnings from compiler

### PR DESCRIPTION
**Pull request by sunreef**
Renamed `ExampleState` to `MainGameState` and moved it to its own file.

Cleaned up the `main.rs` file to avoid it getting too cluttered.

Removed `use` statements and made various modifications to silence compiler warnings.

### CODE REVIEW

File: src/systems/combat.rs [hunk 1]
```diff
@@ -83,7 +84,10 @@ impl<'s> System<'s> for PerformDefaultAttackSystem {
             }
 
             if let Some(value) = cooldown {
-                cooldowns.insert(event.attacker, value);
```
Comment (marotili) [inline src/systems/combat.rs:86]: Looking at the amethyst source code, it seems like an error cannot happen.
There it is implemented as `insert(....).expect("unreachable: We are inserting");` I think we should copy that pattern here.

**Review by marotili — APPROVED** [review]

**Comment by marotili** [comment]
Very nice, thank you!
We will soon set up a CI to keep up the code quality 💪

File: src/systems/combat.rs (hunk 1 above)
Comment (marotili) [reply src/systems/combat.rs:86]: @sunreef would you like to make this change or should I merge as it is right now?

Comment (sunreef) [reply src/systems/combat.rs:86]: @marot I changed it and commited.

Comment (marotili) [reply src/systems/combat.rs:86]: Thanks!

**Pull request merged**